staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -55,13 +55,12 @@ func WithAudit(handler http.Handler, sink audit.Sink, policy audit.PolicyRuleEva
 			handler.ServeHTTP(w, req)
 			return
 		}
-		ev := &ac.Event
 
 		ctx := req.Context()
 		omitStages := ac.RequestAuditConfig.OmitStages
 
-		ev.Stage = auditinternal.StageRequestReceived
-		if processed := processAuditEvent(ctx, sink, ev, omitStages); !processed {
+		ac.SetEventStage(auditinternal.StageRequestReceived)
```

Comment:
Setting the event stage is particular to the point it's being output at. Make the stage a parameter to `processAuditEvent`, and set the stage as part of that method instead.

Reply:
Done!